crates/editor/src/element.rs

```diff
@@ -649,22 +649,24 @@ impl EditorElement {
             }
 
             #[cfg(target_os = "linux")]
-            if let Some(text) = cx.read_from_primary().and_then(|item| item.text()) {
-                let point_for_position =
-                    position_map.point_for_position(text_hitbox.bounds, event.position);
-                let position = point_for_position.previous_valid;
-
-                editor.select(
-                    SelectPhase::Begin {
-                        position,
-                        add: false,
-                        click_count: 1,
-                    },
-                    cx,
-                );
-                editor.insert(&text, cx);
+            if EditorSettings::get_global(cx).middle_click_paste {
+                if let Some(text) = cx.read_from_primary().and_then(|item| item.text()) {
+                    let point_for_position =
+                        position_map.point_for_position(text_hitbox.bounds, event.position);
+                    let position = point_for_position.previous_valid;
+
+                    editor.select(
+                        SelectPhase::Begin {
+                            position,
+                            add: false,
+                            click_count: 1,
+                        },
+                        cx,
+                    );
+                    editor.insert(&text, cx);
+                }
+                cx.stop_propagation()
```

Comment:
I think this needs to go out of this condition. Now it only has an effect if `middle_click_paste` is enabled. I think that's wrong.

Reply:
The `stop_propagation`?  Wouldn't that eat the middle click event, even though we're not doing anything with it?  

There are two conditions just above where we skip the `stop_propagation`:

```rust
            if !text_hitbox.is_hovered(cx) || editor.read_only(cx) {
                return;
            }
```

I could be wrong but I assumed this means meant if we're not doing anything with the middleclick event then we shouldn't stop the event propagation.

Concretely, wouldn't stopping it potentially interfere with attempts to use it elsewhere in the program?

Reply:
Hmmm, I see what you mean. What I meant is that the previous behavior was that `cx.stop_propagation` was called here, regardless of whether or not something was pasted. And now the behavior is different: `stop_propagation` is only called if `middle_click_paste` is enabled. If you think that's not an issue, then that's fine. Just wanted to double-check.

Reply:
Gotcha!  Yeah I appreciate your double-checking cause I probably could have looked at this more carefully.  After thinking about it some more, I think should be good like this.  Currently the whole section only deals with linux middle click events, which at the moment only drive the paste logic.  If need be, more calls to `stop_propagation` can be added down the line if/when more functionality is implemented here but for now I think this should work good.